tests/test_execution_client.cpp

```diff
@@ -0,0 +1,28 @@
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+#include "sm_tests.h"
+
+namespace ara::exec {
+    /* Mock ExecutionClient for unit testing. */
+    class MockExecutionClient : public ExecutionClient {
+    public:
+        MOCK_METHOD(void, ReportApplicationState, (sm::FunctionGroupStateType state));
+    };
+}
+
+TEST_F(smTests,  testExecutionClient)
+{
+    ara::exec::MockExecutionClient mockEC;
+    mySM.executionClient = &mockEC;
```

Comment:
is this even legal? XD maybe it's better idea to move this to constructor

Reply:
Done